### ISSUE TITLE
executables like 'pip' not using correct python?

### ISSUE THREAD

**Issue by illume** [glyph/venvdotapp#1]
Hello,
[ED: oops. I used `python -m venv` not `virtualenv`]
I noticed an issue where if I install with `pip install oo` the virtualenv can't find the package.
But if I use `python -m pip install oo` it works fine.
```
(anenv363) 16:53:20-rene~$ head -4 /tmp/anenv363/bin/pip
#!/private/tmp/anenv363/bin/python3

# -*- coding: utf-8 -*-
import re
(anenv363) 16:53:26-rene~$ head -4 /tmp/anenv363/bin/python
    #!/bin/bash
    exec '/private/tmp/anenv363/bin/anenv363.app/Contents/MacOS/python' "$@";
```

More details here: https://github.com/pygame/pygame/issues/203#issuecomment-365634298

**Comment by jyhong836** [comment]
I think I find some clues about the issue. The conclusion is

+ `venvdotapp` is not necessary and should not be used with `venv` (in standard lib of python3.3+)

Different mechanism between `venv` and `virtualenv`:
+ `venv` makes links of python executable files.
+ `virtualenv` copies

What does `venvdotapp` do? It encloses `python` binary file in a mac app package or bundle and modifies python file as you see.

Since `venv` only makes a symbol link to the system `python`, it will make no difference to using the system python.

**Comment by jyhong836** [comment]
#### About the problem why `pip` cannot work with `venvdotapp`

Below are all under a virtual environment created by `venv`
1. Before run `venvdotapp`, the python is a link point to the system one: `python --> /usr/local/bin/python3`
2. After run `venvdotapp`, the python is a link chain like: `python --> test_venv.app/Contents/MacOS/python --> /usr/local/.../python3` whose proof:
```
(test_venv) $ venvdotapp
plistloc /Users/.../test_venv/bin/test_venv.app/Contents/Info.plist
apppy /Users/.../test_venv/bin/test_venv.app/Contents/MacOS/python
realpython /usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/bin/python3.6
```
You can find the folder `/Users/.../test_venv/` is not in the link of real python. In contrast, with `virtualenv`, `venvdotapp` will link real python to `/Users/.../test_venv/bin/python3`.
3. When you invoke the new `python`, it will think the python path is `/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/bin/` but not ` /Users/.../test_venv/`. And the new `python` will search the system environment for packages.
4. When you invoke `pip install oo`, the `oo` package will be installed inside ` /Users/.../test_venv/`. That is why the new `python` cannot find installed packages.
5. When you invoke `python -m pip install oo`, the `oo` package is installed in system environment. I think it is not what you expect.

For `venv`, I think the problem `venvdotapp` hopes to address does not exist. Since I do not develop an app with python, I am not sure if it is true for that case. At least, I guess the linked python should work when system python works.

**Comment by jyhong836** [comment]
I think `envdotapp` should alert when `python` is a symbolic link.
Otherwise, users of `venv` will get unexpected results, installing system-wise packages.

**Comment by illume** [comment]
Yeah, sorry about that. I was using `python -m venv`. Thanks for the explanation.

**Comment by glyph** [comment]
I think this should be resolved in the new version 0.1.0, introduced in this commit: eb885058851ff402f086b20ffcfd7303830d21f0 - the new strategy should be considerably more robust.

Let me know if it's still broken for you!
